Dockerfile.updater-core

```diff
@@ -10,6 +10,40 @@ ENV DEBIAN_FRONTEND="noninteractive" \
   LC_ALL="en_US.UTF-8" \
   LANG="en_US.UTF-8"
 
+# Configures Ubuntu to use Azure mirrors, which are often faster and more reliable
+# From https://github.com/Azure/LinuxPatchExtension/blob/master/src/tools/references/apt/sources.list.d/ubuntu.sources
+# This is also used in other places such as the runner builds: https://github.com/actions/runner-images/blob/2f8d6ab9ef24fd8bda80b5259650fe1dc4311320/images/ubuntu/scripts/build/configure-apt-sources.sh#L11
+RUN set -e; \
+    release=$( . /etc/os-release && echo "$VERSION_CODENAME" ); \
+    cat <<EOF | tee /etc/apt/sources.list.d/ubuntu.sources
+
+# Prefer Azure mirrors
+Types: deb
+URIs: http://azure.archive.ubuntu.com/ubuntu/
+Suites: ${release} ${release}-updates ${release}-backports
+Components: main universe restricted multiverse
+
+Types: deb
+URIs: http://azure.archive.ubuntu.com/ubuntu/
+Suites: ${release}-security
+Components: main universe restricted multiverse
```

Comment:
I get a warning when building the image now:

```
N: Missing Signed-By in the sources.list(5) entry for 'http://azure.archive.ubuntu.com/ubuntu'
```

Reply:
Thanks for pointing that out. This was a miss during the refactor. I fixed it 31aa3440d5546d6aa5c0526d1011e2a0897c268d